spinn_front_end_common/interface/interface_functions/front_end_common_insert_live_packet_gatherers_to_graphs.py

```diff
@@ -0,0 +1,191 @@
+# spinn front end common imports
+from pacman.model.graphs.common.slice import Slice
+from spinn_front_end_common.utilities import exceptions
+from spinn_front_end_common.utility_models.live_packet_gather \
+    import LivePacketGather
+from spinn_front_end_common.utility_models.live_packet_gather_machine_vertex\
+    import LivePacketGatherMachineVertex
+
+# pacman imports
+from pacman.model.constraints.placer_constraints\
+    .placer_chip_and_core_constraint import PlacerChipAndCoreConstraint
+from spinn_utilities.progress_bar import ProgressBar
+
+
+class FrontEndCommonInsertLivePacketGatherersToGraphs(object):
+    """ function to add LPG's as required into a given graph
+    """
+
+    def __call__(
+            self, live_packet_gatherers, machine, machine_graph,
+            application_graph=None, graph_mapper=None):
+        """ call that adds LPG vertices on ethernet connected chips as\
+         required.
+
+        :param live_packet_gatherers: the LPG parameters requested by the \
+        script
+        :param machine: the spinnaker machine as discovered
+        :param application_graph: the application graph
+        :param machine_graph: the machine graph
+        :return: mapping between LPG params and LPG vertex
+        """
+        lpg_params_to_vertex_mapping = dict()
+
+        # create progress bar
+        progress_bar = ProgressBar(
```

Comment:
Comments about other progress bar in this review _do not_ apply to this one. This one is fine.